src/main/java/org/broadinstitute/hellbender/engine/FeatureDataSource.java

```diff
@@ -248,8 +267,31 @@ public FeatureDataSource(final FeatureInput<T> featureInput, final int queryLook
             this.hasIndex = false;
             this.supportsRandomAccess = true;
         } else if (featureReader instanceof AbstractFeatureReader) {
+
             this.hasIndex = ((AbstractFeatureReader<T, ?>) featureReader).hasIndex();
             this.supportsRandomAccess = hasIndex;
+
+            // Check that our index is not out-of-date:
+            if( hasIndex ) {
+                final Path featureFilePath = IOUtils.getPath( featureInput.getFeaturePath() );
+
+                if ( !Files.exists(featureFilePath) ) {
+                    throw new GATKException( "Index file should exist, but unable to find it!  Unable to check if index is out of date!" );
```

Comment:
This checks if featureFilePath exists and then if it throws it says it can't find the index, seems a bit strange.  You should also include the path to the file that it can't find in the error message because otherwise it's hard to know what it was looking for.  

This should probably be a user exception as well, usually missing files are the users fault.

Reply:
Fixed!